filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -189,10 +189,17 @@ func startHarvester(
 
 		harvesterCtx, cancelHarvester, err := hg.readers.newContext(srcID, canceler)
 		if err != nil {
-			// The returned may or not be collected by the caller, thus logging
-			// it here is important.
-			ctx.Logger.Errorf("error while adding new reader to the bookkeeper %v", err)
-			return fmt.Errorf("error while adding new reader to the bookkeeper %w", err)
+			// The only possible returned error is ErrHarvesterAlreadyRunning, which is a normal
+			// behaviour of the Filestream input, it's not really an error, it's just an situation.
+			// If the harvester is already running we don't need to start a new one.
+			// At the moment of writing even the returned error is ignored. So the
+			// only real effect of the next line is to not start a second harvester.
+			//
+			// Currently the only places this error is checked is on task.Group and the
+			// only thing it does is to log the error. So to avoid unnecessary errors,
+			// we just return nil.
+			//nolint: nilerr // Read the comment above
+			return nil
```

Comment:
> // The only possible returned error is ErrHarvesterAlreadyRunning

The only possible returned error *today* is ErrHarvesterAlreadyRunning.  Perhaps we should add a case that checks for this error specifically, and return `nil` as you have done here?  And then for any other errors (future-proofing), it could keep the old behavior?

Reply:
Done on https://github.com/elastic/beats/pull/35426/commits/fec14207b94d1a7d6ff4c1179839ce8611c0ff36